### PR TITLE
fix(template): remove sub from metadata

### LINKED ISSUE #4560: Unable to deploy service with storage
I recently updated my Copilot CLI to v1.26.0 and started a new application from scratch. 

After initiating the storage, I'm unable to deploy the service:

```shell
$ copilot storage init -n assets -t S3 -w pspdfkit
$ copilot env deploy
$ copilot svc deploy --name pspdfkit
Only found one environment, defaulting to: eu
✔ Proposing infrastructure changes for stack pdf-eu-pspdfkit
✘ deploy service pspdfkit to environment eu: deploy service: parse cloudformation template for resource descriptions: decode resource Metadata for description: yaml: unmarshal errors:
  line 14: cannot unmarshal !Sub `` into string
```

The problem appears to be caused by the format of the generated access policy addon in `copilot/pspdfkit/addons/assets-access-policy.yml`:

```yaml
Parameters:
  App:
    Type: String
    Description: Your application's name.
  Env:
    Type: String
    Description: The environment name your service, job, or workflow is being deployed to.
  Name:
    Type: String
    Description: Your workload's name.
Resources:
  assetsBucketAccessPolicy:
    Metadata:
      'aws:copilot:description': !Sub
        - 'An IAM managed policy for your service to access the ${Bucket} bucket of  your environment'
        - Bucket: { Fn::ImportValue: { Fn::Sub: "${App}-${Env}-assetsBucketName" }}
    Type: AWS::IAM::ManagedPolicy
    Properties:
      Description: !Sub
        - Grants CRUD access to the S3 bucket ${Bucket}
        - Bucket: { Fn::ImportValue: { Fn::Sub: "${App}-${Env}-assetsBucketName" }}
      PolicyDocument:
        Version: '2012-10-17'
        Statement:
          - Sid: S3ObjectActions
            Effect: Allow
            Action:
              - s3:GetObject
              - s3:PutObject
              - s3:PutObjectACL
              - s3:PutObjectTagging
              - s3:DeleteObject
              - s3:RestoreObject
            Resource: !Sub
              - ${ BucketARN }/*
              - BucketARN: { Fn::ImportValue: { Fn::Sub: "${App}-${Env}-assetsBucketARN" }}
          - Sid: S3ListAction
            Effect: Allow
            Action: s3:ListBucket
            Resource:
              Fn::ImportValue: !Sub "${App}-${Env}-assetsBucketARN"

Outputs:
  assetsNameBucketName:
    # Injected as ASSETS_NAME_BUCKET_NAME environment variable into your main container.
    Description: "The name of a user-defined bucket."
    Value: { Fn::ImportValue: { Fn::Sub: "${App}-${Env}-assetsBucketName" }}
  assetsBucketAccessPolicy:
    Description: "The IAM::ManagedPolicy to attach to the task role"
    Value: !Ref assetsBucketAccessPolicy
```

If I comment the `Resources->assetsBucketAccessPolicy->Metadata` block`, I can deploy the service.

Any idea why the Copilot CLI is unable to parse the YAML generated by itself?

### PR DESCRIPTION
**Pull request by iamhopaul123**
fixes https://github.com/aws/copilot-cli/issues/4560

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/template/templates/addons/s3/env/access_policy.yml [hunk 1]
```diff
@@ -11,9 +11,7 @@ Parameters:
 Resources:
   {{logicalIDSafe .Name}}BucketAccessPolicy:
     Metadata:
-      'aws:copilot:description': !Sub
```
Comment (efekarakus) [inline internal/pkg/template/templates/addons/s3/env/access_policy.yml:14]: Is this the only template with !Sub? in the metadata description

**Review by efekarakus — APPROVED** [review]

File: internal/pkg/template/templates/addons/s3/env/access_policy.yml (hunk 1 above)
Comment (iamhopaul123) [reply internal/pkg/template/templates/addons/s3/env/access_policy.yml:14]: actually no

File: internal/pkg/template/templates/addons/ddb/env/access_policy.yml [hunk 2]
```diff
@@ -11,9 +11,7 @@ Parameters:
 Resources:
   {{logicalIDSafe .Name}}DdbTableAccessPolicy:
     Metadata:
-      'aws:copilot:description': !Sub
-        - 'An IAM managed policy for your service to access the ${Table} DDB table of your environment'
-        - Table: { Fn::ImportValue: { Fn::Sub: "${App}-${Env}-{{logicalIDSafe .Name}}TableName" } }
+      'aws:copilot:description': 'An IAM managed policy for your service to access the DDB table of your environment'
```
Comment (huanjani) [inline internal/pkg/template/templates/addons/ddb/env/access_policy.yml]: What if there is more than one DDB table in the env? Does that matter? 🤔 "...access **a** DDB table **in**..."??

**Review by huanjani — APPROVED** [review]

File: internal/pkg/template/templates/addons/ddb/env/access_policy.yml (hunk 2 above)
Comment (iamhopaul123) [reply internal/pkg/template/templates/addons/ddb/env/access_policy.yml]: good call

**Pull request merged**